apollo-mockserver/src/commonMain/kotlin/com/apollographql/apollo3/mockserver/MockServerCommon.kt

```diff
@@ -51,10 +50,15 @@ class MockResponse(
   constructor(
       body: String,
       statusCode: Int = 200,
-      headers: Map<String, String> = emptyMap()
+      headers: Map<String, String> = emptyMap(),
   ) : this(statusCode, body.encodeUtf8(), headers)
 }
 
+interface MockDispatcher {
+  fun dispatch(request: MockRecordedRequest): MockResponse
```

Comment:
Naming: it feels weird having `"Recorded"` in this API as it's not really recorded yet. What about introducing `MockRequest` ?

Reply:
Agreed, `MockRequest` makes more sense!